test/jdk/javax/swing/JSplitPane/TestSplitPaneEnableTest.java

```diff
@@ -0,0 +1,150 @@
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 5021949
+ * @key headful
+ * @summary  Verifies JSplitPane setEnabled(false) disables one touch expandable clicks
+ * @run main TestSplitPaneEnableTest
+ */
+
+import java.awt.Point;
+import java.awt.Robot;
+import java.awt.event.InputEvent;
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JSplitPane;
+import javax.swing.plaf.basic.BasicSplitPaneDivider;
+import javax.swing.plaf.basic.BasicSplitPaneUI;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import javax.swing.UnsupportedLookAndFeelException;
+
+public class TestSplitPaneEnableTest {
+    private static JButton leftOneTouchButton;
+    private static JButton rightOneTouchButton;
+    private static JFrame frame;
+    private static JSplitPane jsp;
+    private static volatile Point loc;
+    private static volatile int prevDivLoc;
+    private static volatile int curDivLoc;
+
+    private static void setLookAndFeel(UIManager.LookAndFeelInfo laf) {
+        try {
+            UIManager.setLookAndFeel(laf.getClassName());
+        } catch (UnsupportedLookAndFeelException ignored) {
+            System.out.println("Unsupported LAF: " + laf.getClassName());
+        } catch (ClassNotFoundException | InstantiationException
+                 | IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static void main(String[] args) throws Exception {
+        Robot robot = new Robot();
+        for (UIManager.LookAndFeelInfo laf : UIManager.getInstalledLookAndFeels()) {
+            if (laf.getClassName().toLowerCase().contains("gtk")) {
+                continue;
+            }
+            System.out.println("Testing LAF : " + laf.getClassName());
+            SwingUtilities.invokeAndWait(() -> setLookAndFeel(laf));
+
+            try {
+                SwingUtilities.invokeAndWait(() -> {
+                    frame = new JFrame("SplitPaneTest");
+                    jsp = new JSplitPane(JSplitPane.HORIZONTAL_SPLIT,
+                                         new JButton("Left"),
+                                         new JButton("Right"));
+
+                    frame.add(jsp);
+                    jsp.setUI(new TestSplitPaneUI());
+                    jsp.setOneTouchExpandable(true);
+
+                    jsp.setEnabled(false);
+
+                    frame.setSize(400, 200);
+                    frame.setLocationRelativeTo(null);
+                    frame.setVisible(true);
+                });
+
+                robot.waitForIdle();
+                robot.delay(1000);
+
+                SwingUtilities.invokeAndWait(() -> {
+                    loc = leftOneTouchButton.getLocationOnScreen();
+                    prevDivLoc = jsp.getDividerLocation();
+                });
+                robot.mouseMove(loc.x, loc.y);
+                robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+                robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
```

Comment:
Isn't it enough to verify the state of the buttons whether they're enabled or not. It should be simpler, it could be done without rendering the frame.

Reply:
Ok..modified to just check for button state...